fluss-server/src/main/java/com/alibaba/fluss/server/coordinator/LakeTableTieringManager.java

```diff
@@ -0,0 +1,426 @@
+
+    private final Timer lakeTieringScheduleTimer;
+    private final Clock clock;
+    private final Queue<Long> pendingTieringTables;
+    private final LakeTieringExpiredOperationReaper expirationReaper;
+
+    // the tiering state of the table to be tiered,
+    // from table_id -> tiering state
+    private final Map<Long, TieringState> tieringStates;
+
+    // the live tables that are tiering,
+    // from table_id -> last heartbeat time by the tiering service
+    private final Map<Long, Long> liveTieringTableIds;
+
+    // table_id -> table path
+    private final Map<Long, TablePath> tablePathById;
+
+    // table_id -> tiering interval
+    private final Map<Long, Long> tieringIntervalByTableId;
+
+    private final Lock lock = new ReentrantLock();
+
+    public LakeTableTieringManager() {
+        this(new DefaultTimer("delay lake tiering", 1_000, 20), SystemClock.getInstance());
+    }
+
+    @VisibleForTesting
+    protected LakeTableTieringManager(Timer lakeTieringScheduleTimer, Clock clock) {
+        this.lakeTieringScheduleTimer = lakeTieringScheduleTimer;
+        this.clock = clock;
+        this.pendingTieringTables = new ArrayDeque<>();
+        this.tieringStates = new HashMap<>();
+        this.liveTieringTableIds = new HashMap<>();
+        this.tablePathById = new HashMap<>();
+        this.tieringIntervalByTableId = new HashMap<>();
+        this.expirationReaper = new LakeTieringExpiredOperationReaper();
+        expirationReaper.start();
+    }
+
+    public void initWithLakeTables(List<TableInfo> lakeTables) {
+        for (TableInfo tableInfo : lakeTables) {
+            doHandleStateChange(tableInfo.getTableId(), TieringState.ReInitialized);
+            addLakeTableToTier(tableInfo);
+        }
+    }
+
+    public void addNewLakeTable(TableInfo tableInfo) {
+        doHandleStateChange(tableInfo.getTableId(), TieringState.New);
+        addLakeTableToTier(tableInfo);
+    }
+
+    private void addLakeTableToTier(TableInfo tableInfo) {
+        inLock(
+                lock,
+                () -> {
+                    tablePathById.put(tableInfo.getTableId(), tableInfo.getTablePath());
+                    tieringIntervalByTableId.put(
+                            tableInfo.getTableId(),
+                            tableInfo.getTableConfig().getDataLakeTieringInterval().toMillis());
+                    doHandleStateChange(tableInfo.getTableId(), TieringState.Scheduled);
+                });
+    }
+
+    private void scheduleTableTiering(long tableId) {
+        scheduleTableTiering(tableId, null);
+    }
+
+    private void scheduleTableTiering(long tableId, @Nullable Long delayMs) {
+        Long tieringInterval = tieringIntervalByTableId.get(tableId);
+        if (tieringInterval == null) {
+            // the table has been dropped, return directly
+            return;
+        }
+        lakeTieringScheduleTimer.add(
```

Comment:
We should add it into the tiering queue directly if the last committed snapshot exceeds the tiering interval. Currently, if the coordinator restart, it adds a tiering interval to all the tables which is not necessary.

Reply:
Yes, it was just to simplify the logic. But seems add the logic is not complex, now, I added the logic in the TieringManager but left add snapshot commit timestamp in next pr.